### PR TITLE
Add Xbox documentation

### PR DESCRIPTION
**Pull request by hunterjm**
## Proposed change

Add documentation for the Xbox Integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/41697
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/1928
- This PR fixes or closes issue: N/A

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/xbox.markdown
Comment (hunterjm) [inline source/_integrations/xbox.markdown]: I wasn't sure what to do here because I plan to have this available through cloud linking, so didn't want to force users through the below steps, but it doesn't look like anything else that uses account linking mentions it in the docs?

**Comment by MartinHjelmare** [comment]
Should we mention what consoles are supported? It looks like it's only Xbox One and later?

File: source/_integrations/xbox.markdown
Comment (MartinHjelmare) [reply source/_integrations/xbox.markdown]: Looks good to me! Both ways should be documented.

File: source/_integrations/xbox.markdown
Comment (MartinHjelmare) [inline source/_integrations/xbox.markdown]:
```suggestion
Just like the binary sensors, the Xbox sensor platform automatically keeps track of your "**Favorite** friends".
```

**Comment by hunterjm** [comment]
Ok, I'm done adding changes for 0.117 :)

File: source/_integrations/xbox.markdown
Comment (MartinHjelmare) [reply source/_integrations/xbox.markdown]: What's the status for the account linking? This isn't available yet?

Comment (hunterjm) [reply source/_integrations/xbox.markdown]: I’ve sent @balloob the info. Haven’t heard anything yet.

Comment (MartinHjelmare) [reply source/_integrations/xbox.markdown]: So maybe we shouldn't mention this until that is ready?

Comment (hunterjm) [reply source/_integrations/xbox.markdown]: Sorry. By “I haven’t heard anything”, I meant I haven’t heard about it being done. He did confirm that it should be no problem to get done and requested the steps to get the client credentials.

Comment (MartinHjelmare) [reply source/_integrations/xbox.markdown]: Ok, but then we're blocking this PR until confirmation that it's done. I was thinking it's better to unblock the PR and make another PR when it's ready.

**Comment by SeanPM5** [comment]
For the media source section, it might be worth including a brief mention about potential delays in newly downloaded content appearing in the Media Browser. 

When I downloaded a few extra media player apps on my console to test this integration, it took ~21 hours for them to appear in HA. Not sure if that's a consistent behavior though...

File: source/_integrations/xbox.markdown
Comment (hunterjm) [reply source/_integrations/xbox.markdown]: Circled up with Paulus and he said he would do it on Monday.

Comment (hunterjm) [reply source/_integrations/xbox.markdown]: @MartinHjelmare - Account Linking is now configured and verified working.

**Review by MartinHjelmare — DISMISSED** [review]

**Comment by hunterjm** [comment]
LGTM

**Review by MartinHjelmare — DISMISSED** [review]

**Review by frenck — APPROVED** [review]
Thanks, @hunterjm 👍 

Awesome work 🎉

**Pull request merged**
